cpp/src/search/contains.cu

```diff
@@ -107,10 +100,11 @@ bool contains_scalar_dispatch::operator()<cudf::dictionary32>(column_view const&
 
 struct multi_contains_dispatch {
   template <typename Type>
-  std::unique_ptr<column> operator()(column_view const& haystack,
-                                     column_view const& needles,
-                                     rmm::cuda_stream_view stream,
-                                     rmm::mr::device_memory_resource* mr) const
+  std::enable_if_t<!is_nested<Type>(), std::unique_ptr<column>> operator()(
```

Comment:
I would try to use `CUDF_ENABLE_IF` throughout this PR.

To my knowledge, there is no significant difference between
```cpp
template<std::enable_if_t<condition>* = nullptr>
return_type function_name(...)
```
and
```cpp
std::enable_if_t<condition, return_type> function_name(...)
```

However, I believe we are aiming to uniformly use the former construction of SFINAE and replace raw `std::enable_if_t` calls with [`CUDF_ENABLE_IF`](https://github.com/rapidsai/cudf/blob/4e66281f48c55735edb4b610e0f859ee2de32a75/cpp/include/cudf/utilities/traits.hpp#L48). See examples: https://github.com/rapidsai/cudf/blob/4e66281f48c55735edb4b610e0f859ee2de32a75/cpp/src/copying/copy.cu#L53

Reply:
Not possible to use `CUDF_ENABLE_IF` without more changes to the code: the `__T291` club prevent me to do that.
Ref: https://nvidia.slack.com/archives/CDTANRCTT/p1616429054078600

Reply:
For the public record, the short description of that Slack conversation is that nvcc has some restrictions on where `__device__` lambdas can be used. This is a case that fails with an error like `Can't compile: error: '__T291' was not declared in this scope`. "The fix is: pass a functor instead of the lambda."

@ttnghia Is it possible to do the same trick here? Is there a specific device lambda that could be changed to a functor to avoid this problem?

Reply:
The problem is due to `std::enable_if` used inside the template bracket (`template<typename T, std::enable_if....>`).

I hacked around to use `CUDF_ENABLE_IF` with functor, eventually. Please check.

Reply:
I'm generally happy with the changes in 385fa1d713a9429ba709ea1c0be4888a1b3dfb55! I have some further questions/suggestions about the changes but will discuss in separate threads. Resolving.